src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/SelectExpandNode.cs

```diff
@@ -129,8 +129,8 @@ public SelectExpandNode(SelectExpandClause selectExpandClause, IEdmStructuredTyp
                 }
                 else
                 {
-                    BuildSelections(selectExpandClause, allStructuralProperties, allComplexStructuralProperties, allNavigationProperties, allActions, allFunctions);
                     SelectAllDynamicProperties = false;
+                    BuildSelections(selectExpandClause, allStructuralProperties, allComplexStructuralProperties, allNavigationProperties, allActions, allFunctions);
```

Comment:
Does BuildSelections modify SelectAllDynamicProperties? I am wondering why the function call is moved after

Reply:
Yes. In the BuildSelections(…), we should set it as "true" if we have `$select=*`. 

I have to move the initialization ahead of "BuildSelections(...)" method call.

Reply:
Okay, I think the change you have is the quickest and easiest way to fix it, but I'm not comfortable with the way we are updating SelectAllDynamicProperties since we are setting it to false and then _maybe_ BuildSelections will update it to true. Would it be a lot of code changes to add an additional `out` boolean parameter to BuildSelections? And then set SelectAllDynamicProperties to the boolean? I think that would be an easier code pattern to maintain and read.

Reply:
Emm, well, it's not i think. It looks to initialize a default value to "SelectAllDynamicProperties" property. It should move to the head of this constructor or in its property initilizer. But we put here it to make sure we initialize it as a correct value.

Reply:
As we discussed, perhaps we can comment that BuildSelections modifies the SelectAllDynamicProperties to true in some conditions.